services/src/contexts/mod.rs

```diff
@@ -80,7 +80,8 @@ pub trait SessionContext: 'static + Send + Sync + Clone {
     fn tasks(&self) -> Self::TaskManager;
 
     /// Create a new query context for executing queries on processors
-    fn query_context(&self) -> Result<Self::QueryContext>;
+    // TODO: assign computation id inside SessionContext or let it be provided from outside?
```

Comment:
I guess this is ok?

Reply:
Ich weiß nicht, Dependency Injection klingt schon ganz gut, vielleicht sollte man eine Factory reingeben statt einer Workflow ID